ddtrace/contrib/botocore/services/bedrock.py

```diff
@@ -240,23 +214,18 @@ def _extract_streamed_response(span: Span, streamed_body: List[Dict[str, Any]])
         elif provider == _COHERE and streamed_body:
             if "is_finished" in streamed_body[0]:  # streamed response
                 if "index" in streamed_body[0]:  # n >= 2
-                    n = int(span.get_tag("bedrock.request.n") or 0)
+                    num_generations = int(ctx.get_item("num_generations") or 0)
                     text = [
                         "".join([chunk["text"] for chunk in streamed_body[:-1] if chunk["index"] == i])
-                        for i in range(n)
+                        for i in range(num_generations)
                     ]
-                    finish_reason = [streamed_body[-1]["finish_reason"] for _ in range(n)]
+                    finish_reason = [streamed_body[-1]["finish_reason"] for _ in range(num_generations)]
                 else:
                     text = "".join([chunk["text"] for chunk in streamed_body[:-1]])
                     finish_reason = streamed_body[-1]["finish_reason"]
             else:
                 text = [chunk["text"] for chunk in streamed_body[0]["generations"]]
                 finish_reason = [chunk["finish_reason"] for chunk in streamed_body[0]["generations"]]
-                for i in range(len(text)):
-                    span.set_tag_str(
-                        "bedrock.response.choices.{}.id".format(i),
-                        str(streamed_body[0]["generations"][i].get("id", None)),
-                    )
```

Comment:
Just to confirm, was this (streamed case) and the non-streamed case combined to set `body` and handle both in the same place in `botocore.bedrock.process_response`?

Reply:
Yup, I was able to deduplicate some similar bits of code by the implementation of `_on_botocore_bedrock_process_response`.